indico/modules/rb_new/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -0,0 +1,659 @@
+    }
+
+    async fetchRoomAttributes() {
+        const {roomId} = this.props;
+        let response;
+        try {
+            response = await indicoAxios.get(fetchRoomAttributesURL({room_id: roomId}));
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        this.setState({roomAttributes: response.data});
+    }
+
+    async fetchAttributes() {
+        let response;
+        try {
+            response = await indicoAxios.get(fetchAttributesURL());
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        this.setState({attributes: response.data});
+    }
+
+
+    async fetchRoomAvailability() {
+        const {roomId} = this.props;
+        let response;
+        try {
+            response = await indicoAxios.get(fetchRoomAvailabilityURL({room_id: roomId}));
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        const roomAvailability = camelizeKeys(response.data);
+        roomAvailability.nonbookablePeriods.forEach(period => {
+            period.key = shortid.generate();
+        });
+        roomAvailability.bookableHours.forEach(hours => {
+            hours.key = shortid.generate();
+        });
+
+        this.setState({roomAvailability});
+    }
+
+    async fetchRoomEquipment() {
+        const {roomId} = this.props;
+        let response;
+        try {
+            response = await indicoAxios.get(fetchRoomEquipmentURL({room_id: roomId}));
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        this.setState({roomEquipment: camelizeKeys(response.data)});
+    }
+
+    handleCloseModal = () => {
+        const {onClose, roomId, actions: {fetchRoom, fetchRoomDetails}} = this.props;
+        fetchRoom(roomId);
+        fetchRoomDetails(roomId, true);
+        onClose();
+    };
+
+    handleSubmit = async (data, form) => {
+        const {roomId} = this.props;
+        const changedValues = getChangedValues(data, form);
+        const basicDetails = _.omit(changedValues, ['attributes', 'bookableHours', 'nonbookablePeriods', 'availableEquipment']);
+        if ('owner' in changedValues) {
+            basicDetails.owner_id = changedValues.owner.id;
+        }
+        const {availableEquipment, nonbookablePeriods, bookableHours, attributes} = changedValues;
+
+        try {
+            await this.saveBasicDetails(roomId, basicDetails);
+            await this.saveEquipment(roomId, availableEquipment);
+            await this.saveAttributes(roomId, attributes);
+            await this.saveAvailability(roomId, changedValues, nonbookablePeriods, bookableHours);
+            this.setState({submitSucceeded: true});
+        } catch (e) {
+            handleAxiosError(e);
+            this.setState({submitSucceeded: false});
+        }
+    };
+
+    saveBasicDetails = (roomId, basicDetails) => {
```

Comment:
I think that you could make all those `save*` functions `async` and then just `return await ...` in them

Reply:
I just noticed that all those functions are already async and I am returning await, so I was using await twice. Thanks for pointing that out!